tests/legacy-cli/e2e/tests/commands/add/npm-env-vars.ts

```diff
@@ -0,0 +1,35 @@
+import { expectFileNotToExist, expectFileToExist, writeFile } from '../../../utils/fs';
+import { getActivePackageManager, setRegistry } from '../../../utils/packages';
+import { git, ng } from '../../../utils/process';
+import {
+  createNpmConfigForAuthentication,
+  resetNpmEnvVars,
+  setNpmEnvVarsForAuthentication,
+} from '../../../utils/registry';
+
+export default async function () {
+  const packageManager = getActivePackageManager();
+
+  if (packageManager === 'npm') {
+    try {
+      const command = ['add', '@angular/pwa', '--skip-confirmation'];
+
+      // Environment variables only
+      await expectFileNotToExist('src/manifest.webmanifest');
+      await writeFile('.npmrc', '');
+      setNpmEnvVarsForAuthentication();
+      await ng(...command);
+      await expectFileToExist('src/manifest.webmanifest');
+      await git('clean', '-dxf');
+
+      // Mix of config file and env vars works
+      await expectFileNotToExist('src/manifest.webmanifest');
+      await createNpmConfigForAuthentication(false, true);
+      await ng(...command);
+      await expectFileToExist('src/manifest.webmanifest');
+    } finally {
+      await setRegistry(true);
```

Comment:
```suggestion
```

Reply:
This is one of the changes I found was needed because subsequent tests fail due to their assumption that the *.npmrc* file is pointing to the non-authenticated verdaccio server (:4873) but it's still set to :4876 when it leaves this test.  I added this step to set the environment back to its original state prior to my test.

Reply:
Files get cleaned prior every test run. https://github.com/angular/angular-cli/blob/4a08ec07913ecb3f8b981d6439f1bbfba86221f4/tests/legacy-cli/e2e_runner.ts#L182

Reply:
Gotcha.  Removing...